bigtable/spark/build.sbt

```diff
@@ -0,0 +1,76 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+name := "bigtable-spark-samples"
+
+version := "0.1"
+
+// Versions to match Dataproc 1.4
+// https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-1.4
+scalaVersion := "2.11.12"
+val sparkVersion = "2.4.6"
+val bigtableVersion = "1.16.0"
+val hbaseVersion = "1.3.6"
+
+libraryDependencies ++= Seq(
+  "org.apache.spark" %% "spark-sql" % sparkVersion % Provided,
+  "org.apache.hbase.connectors.spark" % "hbase-spark" % "1.0.0" % Provided,
+  "com.google.cloud.bigtable" % "bigtable-hbase-2.x-hadoop" % bigtableVersion
+)
+
+val scalatestVersion = "3.2.0"
+libraryDependencies += "org.scalactic" %% "scalactic" % scalatestVersion
+libraryDependencies += "org.scalatest" %% "scalatest" % scalatestVersion % "test"
+test in assembly := {}
+
+val fixes = Seq(
+  // Required by 'value org.apache.hadoop.hbase.spark.HBaseContext.dstream'
+  "org.apache.spark" %% "spark-streaming" % sparkVersion % Provided,
+  // hbase-server is needed because HBaseContext references org/apache/hadoop/hbase/fs/HFileSystem
+  // hbase-client is declared to override the version of hbase-client declared by bigtable-hbase-2.x-hadoop
+  "org.apache.hbase" % "hbase-server" % hbaseVersion,
+  "org.apache.hbase" % "hbase-client" % hbaseVersion
+)
+libraryDependencies ++= fixes
+
+// Fix for Exception: Incompatible Jackson 2.9.2
+// Version conflict between HBase and Spark
+// Forcing the version to match Spark
+dependencyOverrides += "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.9.10"
+
+// Excluding duplicates for the uber-jar
+// There are other deps to provide necessary packages
+excludeDependencies ++= Seq(
+  ExclusionRule(organization = "asm", "asm"),
+  ExclusionRule(organization = "commons-beanutils", "commons-beanutils"),
+  ExclusionRule(organization = "commons-beanutils", "commons-beanutils-core"),
+  ExclusionRule(organization = "org.mortbay.jetty", "servlet-api")
+)
+
+assemblyMergeStrategy in assembly := {
+  case PathList("META-INF", "io.netty.versions.properties") => MergeStrategy.first
+  case PathList("META-INF", "MANIFEST.MF") => MergeStrategy.discard
+  case PathList("mozilla", "public-suffix-list.txt") => MergeStrategy.first
+  case PathList("google", xs @ _*) => xs match {
+    case ps @ (x :: xs) if ps.last.endsWith(".proto") => MergeStrategy.first
+    case _ => MergeStrategy.deduplicate
+  }
+  case x =>
+    val oldStrategy = (assemblyMergeStrategy in assembly).value
+    oldStrategy(x)
+    // FIXME Make sure first is OK (it's worked well so far)
+    MergeStrategy.first
```

Comment:
This might cause issues with guava, since our libraries use the most up-to-date version and spark (and especially HBase) doesn't necessarily track as close there is a chance for a diamond. It might be worth asking tomo how we might run linkage checker against an assembly and seeing if it would work.

Reply:
We try to keep the dependency versions in line with HBase for the adapter, so this should be ok.